src/Test/L0/Plugin/FingerprintCreatorTests.cs

```diff
@@ -0,0 +1,177 @@
+                Assert.Equal(FingerprintCreator.SummarizeString($"\nSHA256({path2})=[{content2.Length}]{hash2.ToHex()}"), f.Segments[1]);
+            }
+        }
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Plugin")]
+        public void Fingerprint_FileRelative()
+        {
+            string workingDir = Path.GetDirectoryName(path1);
+            string relPath1 = Path.GetFileName(path1);
+            string relPath2 = Path.GetFileName(path2);
+
+            using(var hostContext = new TestHostContext(this))
+            {
+                var context = new AgentTaskPluginExecutionContext(hostContext.GetTrace());
+                context.SetVariable(
+                    "system.defaultworkingdirectory", // Constants.Variables.System.DefaultWorkingDirectory
+                    workingDir,
+                    isSecret: false);
+
+                var segments = new[]
+                {
+                    $"{relPath1}",
+                    $"{relPath2}",
+                };
+
+                Fingerprint f = FingerprintCreator.ParseFromYAML(context, segments, addWildcard: false);
+                
+                Assert.Equal(2, f.Segments.Length);
+                Assert.Equal(FingerprintCreator.SummarizeString($"\nSHA256({relPath1})=[{content1.Length}]{hash1.ToHex()}"), f.Segments[0]);
+                Assert.Equal(FingerprintCreator.SummarizeString($"\nSHA256({relPath2})=[{content2.Length}]{hash2.ToHex()}"), f.Segments[1]);
+            }
+        }
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Plugin")]
+        public void Fingerprint_Str()
+        {
+            using(var hostContext = new TestHostContext(this))
+            {
+                var context = new AgentTaskPluginExecutionContext(hostContext.GetTrace());
+                var segments = new[]
+                {
+                    $"hello",
+                };
+
+                Fingerprint f = FingerprintCreator.ParseFromYAML(context, segments, addWildcard: false);
+                
+                Assert.Equal(1, f.Segments.Length);
+                Assert.Equal($"hello", f.Segments[0]);
+            }
+        }
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Plugin")]
+        public void Fingerprint_Wildcard()
+        {
+            using(var hostContext = new TestHostContext(this))
+            {
+                var context = new AgentTaskPluginExecutionContext(hostContext.GetTrace());
+                var segments = new[]
+                {
+                    $"hello",
+                };
+
+                Fingerprint f = FingerprintCreator.ParseFromYAML(context, segments, addWildcard: true);
+                
+                Assert.Equal(2, f.Segments.Length);
+                Assert.Equal($"hello", f.Segments[0]);
+                Assert.Equal(Fingerprint.Wildcard, f.Segments[1]);
+            }
+        }
+
+        [Fact]
```

Comment:
@willsmythe Does this cover your compat concerns?

The only scenario where we break is if someone had a single-line key with pipe characters.  Previously that would be treated as a single segment and now it will be multiple segments.